src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildAction.java

```diff
@@ -202,16 +202,20 @@ public static JenkinsBuildInformation retrieveBuildFromSauce(JenkinsSauceREST sa
     }
 
     // Get the list of running jobs and stop them all
-    public void stopJobs() {
+    public void stopJobs() throws InterruptedException {
         JenkinsSauceREST sauceREST = getSauceREST();
         List<JenkinsJobInformation> jobs = getJobs();
-        ExecutorService executor = Executors.newFixedThreadPool(5);
+        List<Future<?>> futures = new ArrayList<>();
         for (JobInformation job : jobs) {
             Runnable worker = new StopJobThread(sauceREST, job);
-            executor.execute(worker);
+            futures.add(Timer.get().submit(worker));
         }
-        executor.shutdown();
-        while (!executor.isTerminated()) {
+        for (Future<?> f : futures) {
+            try {
+                f.get();
```

Comment:
Shouldn't we set a timeout here to avoid still blocking forever?

Reply:
Assuming this is called from a build thread (cannot recall offhand), there is no particular reason to set a timeout—at worst the build will need to be interrupted. Anyway if something in `Timer` blocks for a long time, that is the bigger problem.

Reply:
Yes, I agree, a problem to be solved. My concern was that maybe we are introducing threads hanging a long time because they cannot stop the job.

Reply:
Well if the build is interrupted then `get()` should throw `InterruptedException` I think?

Reply:
Not sure who may interrupt the stopping thread :man_shrugging: Anyway, it's doing things better than before